server/athenian/api/models/web/filtered_code_check_runs.py

```diff
@@ -0,0 +1,79 @@
+from datetime import date
+from typing import List, Optional
+
+from athenian.api.models.web.base_model_ import Model
+from athenian.api.models.web.filtered_code_check_run import FilteredCodeCheckRun
+
+
+class FilteredCodeCheckRuns(Model):
+    """Response from `/filter/code_checks`, found code check runs ordered by repository name and \
+    then by title. Note: we always consider the completed run executions only."""
+
+    openapi_types = {
+        "timeline": List[date],
+        "items": List[FilteredCodeCheckRun],
+    }
+
+    attribute_map = {
+        "timeline": "timeline",
+        "items": "items",
+    }
+
+    def __init__(self,
+                 timeline: Optional[List[date]] = None,
+                 items: Optional[List[FilteredCodeCheckRun]] = None):
+        """FilteredCodeCheckRuns - a model defined in OpenAPI
+
+        :param timeline: The timeline of this FilteredCodeCheckRuns.
+        :param items: The items of this FilteredCodeCheckRuns.
+        """
+        self._timeline = timeline
+        self._items = items
+
+    @property
+    def timeline(self) -> List[date]:
+        """Gets the timeline of this FilteredCodeCheckRuns.
+
+        Sequence of dates from `date_from` till `date_to`. We choose such an interval that
+        the number of items is approximately 50.
```

Comment:
I don't understand what is this? What is the difference with granularities?

Reply:
Consider this the autogranularity that lets us plot everything on X axis. Do you think that the webapp should select how many timestamps? We could agree on a reasonable default and reduce the complexity.

Reply:
I see the goal here is to save us from a massive implementation of the granularities logic, right?
Let's discuss it tomorrow but indeed letting the webapp choosing the number of timestamps might let us approx in a cheap way, the behavior of granularities eg. if date_to - date_from = delta > 28 days the webapp could request delta / 7 timestamps, etc